android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -285,7 +287,9 @@ public void onListItemClick(ListView l, View v, int position, long id) {
       // Initiate WifiDirect Connection request to device
       mWifiDirectConnect.connect(item.getUrlDevice(), pwsResult.getTitle());
     } else if (Utils.isFatBeaconDevice(item.getUrlDevice())) {
-      (new BluetoothSite(getActivity())).connect(pwsResult.getSiteUrl(), pwsResult.getTitle());
+      if (!mBluetoothSite.isRunning()) {
```

Comment:
Can this be added to else if above? 
`else if (Utils.isFatBeaconDevice(item.getUrlDevice() && !mBluetoothSite.isRunning()) {`